app/activity/[id]/index.tsx

```diff
@@ -333,14 +351,20 @@ export default function ActivityDetailScreen() {
                         : 'Payment Transaction'}
                   </ThemedText>
                   <ThemedText style={[styles.infoText, { color: secondaryTextColor }]}>
-                    This was a {isAuth ? 'login' : isTicket ? 'ticket' : 'payment'} request{' '}
-                    {isAuth ? 'to authenticate your identity with' : 'from'} {activity.service_name}
-                    .
+                    {isAuth
+                      ? `This was a login request to authenticate your identity with ${activity.service_name}.`
+                      : isTicket
+                        ? `This was a ticket request${activity.detail ? ` for ${activity.detail}` : ''} from ${activity.service_name}.`
+                        : `This was a payment request from ${activity.service_name}.`}
                     {activityStatus === 'success' &&
                       (isAuth
                         ? ' You successfully granted access.'
                         : isTicket
-                          ? ' You received a ticket.'
+                          ? activity.type === 'ticket_received'
+                            ? ' You successfully received the ticket.'
+                            : activity.type === 'ticket_approved'
+                              ? ' You successfully approved and sent the ticket.'
+                              : ' The ticket was processed successfully.'
```

Comment:
This seems a bit complicated, maybe it makes sense to add some functions with switch/cases inside?